### PR TITLE
Added glbToPlainB3dm to generate basic b3dm from glb

### PR DESCRIPTION
**Pull request by likangning93**
Added a basic tool to attach a basic `b3dm` header to an input `glb` buffer.

### CODE REVIEW

**Comment by pjcozzi** [comment]
Can you also expose a cli (command-line interface) and document it.  For an example, see #4.

File: tools/lib/glbToPlainB3dm.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+module.exports = glbToPlainB3dm;
```
Comment (pjcozzi) [inline tools/lib/glbToPlainB3dm.js]: Can we name this `glbToB3dm`?

File: tools/lib/glbToPlainB3dm.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * @param {Buffer} glbBuffer a buffer representing a binary gltf
+ * @returns {Buffer}
+ */
+function glbToPlainB3dm(glbBuffer) {
+    var header = new Buffer(20);
+    header.write('b3dm', 0); // magic
+    header.writeUInt32LE(1, 4); // version
+    header.writeUInt32LE(glbBuffer.length + 20, 8); // length of entire tile, including header
+    header.writeUInt32LE(0, 12); // number of models in the batch (0 for basic, no batches)
```
Comment (pjcozzi) [inline tools/lib/glbToPlainB3dm.js]: I think this and below need to be swapped for the latest version of b3dm, see https://github.com/AnalyticalGraphicsInc/cesium/pull/4228/files#diff-606f546d0d416eb075bc24aa053eb5ccR211

**Comment by lilleyse** [comment]
> Can you also expose a cli (command-line interface) and document it. For an example, see #4.

@likangning93 I'll try to finish up that PR asap so that it's easier to work with the existing CLI.

File: tools/lib/glbToPlainB3dm.js (hunk 2 above)
Comment (pjcozzi) [inline tools/lib/glbToPlainB3dm.js]: Also, that PR should be merged into `3d-tiles` maybe this afternoon or early tomorrow.

**Comment by lilleyse** [comment]
Er.. I mean this PR: https://github.com/AnalyticalGraphicsInc/3d-tiles-tools/pull/3

File: tools/lib/glbToPlainB3dm.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+module.exports = glbToPlainB3dm;
+
+/**
+ * Generates a new Buffer representing a b3dm file with a plain header.
+ *
+ * @param {Buffer} glbBuffer a buffer representing a binary gltf
+ * @returns {Buffer}
+ */
+function glbToPlainB3dm(glbBuffer) {
```
Comment (pjcozzi) [inline tools/lib/glbToPlainB3dm.js]: throw `DeveloperError` if `glbBuffer` is not defined.

**Comment by pjcozzi** [comment]
I suspect that this simple tool will become surprisingly useful for testing/learning.

@lilleyse review and merge when ready.

**Comment by likangning93** [comment]
This should be ready for another look. Sorry for the delay.

**Comment by likangning93** [comment]
@lilleyse are we waiting on anything else before wrapping this up?

**Comment by lilleyse** [comment]
I'm waiting for #3 to merge first and then see how this PR integrates into that.

File: tools/README.md [hunk 4]
```diff
@@ -1 +1,14 @@
-Command line utilities for processing and converting 3D Tiles tilesets.
\ No newline at end of file
+Command line utilities for processing and converting 3D Tiles tilesets.
+
+# Generate B3dm from Glb
+
+Creates a b3dm from a glb with an empty batch table.
+
+## Example
+`node ./bin/generateB3dmFromGlb -i someGlb.glb -o someB3dm.b3dm`
```
Comment (lilleyse) [inline tools/README.md]: Maybe tweak this example so it uses files that exist in the specs/data folder. The README in master has some examples.

File: tools/README.md [hunk 5]
```diff
@@ -1 +1,14 @@
...
+
+## Command-Line Flags
+| Flag | Description | Required |
+| ---- | ----------- | -------- |
+|`-i` | Input path of the `.glb`| yes |
+|`-o` | Output path of the resulting `.b3dm` | no, defaults to the same as the input path, but with `.b3dm` instead of the pre-existing 3-character extension (typically `glb`)
```
Comment (lilleyse) [inline tools/README.md]: Capitalize Yes and No. I don't think you need the description for 'no'.

File: tools/bin/generateB3dmFromGlb.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+var inputPath = argv.i;
+
+// Default output is just in the same place as input, but with 'b3dm'
+var outputPath = defaultValue(argv.o, inputPath.slice(0, inputPath.length - 3) + 'b3dm');
+
+// Load the glb
+return fsReadFile(inputPath)
+    .then(function(data) {
+        return fsWriteFile(outputPath, glbToB3dm(data));
+    });
```
Comment (lilleyse) [inline tools/bin/generateB3dmFromGlb.js]: Integrate this into `3d-tiles-tools.js` instead.

Also right now we're going with the approach that if the output file doesn't exist we tell the user to use `-f or --force`, so you'll need to check for that.

File: tools/lib/glbToB3dm.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+
+module.exports = glbToB3dm;
+
+/**
+ * Generates a new Buffer representing a b3dm file with a plain header.
```
Comment (lilleyse) [inline tools/lib/glbToB3dm.js]: Change "plain" to "basic" to match your terminology elsewhere.

File: tools/package.json [hunk 8]
```diff
@@ -37,6 +37,7 @@
     "open": "0.0.5",
     "request": "2.74.0",
     "requirejs": "2.2.0",
+    "rimraf": "2.5.4",
```
Comment (lilleyse) [inline tools/package.json]: No need to include this, if you need to remove files you can use `fsExtra.remove` instead.

File: tools/specs/bin/generateB3dmFromGlbSpec.js [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    var generateB3dmFromGlbProcess = childProcess.fork('./bin/generateB3dmFromGlb', args);
+    return new Promise(function(resolve, reject) {
+        generateB3dmFromGlbProcess.on('close', function() {
+            resolve();
+        });
+        generateB3dmFromGlbProcess.on('error', function() {
+            reject();
+        });
+    });
+}
```
Comment (lilleyse) [inline tools/specs/bin/generateB3dmFromGlbSpec.js]: As of now we aren't testing the bin scripts so I would just remove this file.

File: tools/specs/lib/glbToB3dmSpec.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+var glbToB3dm = require('../../lib/glbToB3dm');
```
Comment (lilleyse) [inline tools/specs/lib/glbToB3dmSpec.js]: Place our files after the third party includes.

File: tools/specs/lib/glbToB3dmSpec.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+var fs = require('fs');
+var Promise = require('bluebird');
+
+var fsReadFile = Promise.promisify(fs.readFile);
+
+var glbPath = './specs/data/CesiumTexturedBox/CesiumTexturedBox.glb';
+
+describe('glbToB3dm', function() {
+
```
Comment (lilleyse) [inline tools/specs/lib/glbToB3dmSpec.js]: Remove space.

File: tools/specs/bin/generateB3dmFromGlbSpec.js (hunk 9 above)
Comment (lilleyse) [inline tools/specs/bin/generateB3dmFromGlbSpec.js]: Or... remove for now and when someone writes a test for `3d-tiles-tools.js` this code can be added there.

**Comment by lilleyse** [comment]
Add glbToB3dm to index.js

**Comment by likangning93** [comment]
@lilleyse updated! Sorry for the slow response.

**Comment by lilleyse** [comment]
No problem, everything looks good.

**Pull request merged**